morecantile/models.py

```diff
@@ -1252,9 +1253,11 @@ def tiles(  # noqa: C901
 
         for w, s, e, n in bboxes:
             # Clamp bounding values.
-            w = max(self.bbox.left, w)
+            ws_contain_180th = lons_contain_antimeridian(w, self.bbox.left)
```

Comment:
could we just do a tests to see if the `w > e`

Reply:
The specific case I am hoping to handle is when the `tms.bbox` lons contain the antimeridian, but also the bounds provided for tiles is on either side of the antimeridian or overlaps. This requires we swap the underlying operator (min to max or max to min on a w/e basis) in order to truly clamp and avoid generating many many tiles that don't intersect the provided bounds.

Here is an example: 
Given a `tms.bbox` (110, 0, -158, 90) and bounds used to find tiles of (-179, 45, -178, 46), the final clamped box we want is the bounds itself since it is fully contained inside the `tms.bbox` i.e. (-179, 45, -178, 46). 

- Assuming you mean checking if `w > e` for bounds only: `w > e` evaluates to `False`, and we keep `w = max(self.bbox.left, w)` giving us a new `w` of 110, but we should have -179 so that won't work (hence the need to swap max to min). 
- Assuming you mean checking `w > e` for `tms.bbox` only: `w > e` evaluates to True, so we would swap max to min and get -179, which is what we want! However, consider the same `tms.bbox`, but new bounds that are completely west of the antimeridian (150, 45, 155, 46) and still contained inside `tms.bbox`, then applying the swapped max to min would give us 110, which is not what we want. We would actually need to keep max (not min) here to get the expected 150. That is the reason for needing to check if the anti-meridian is inside interval (`tms.bbox.west`, bounds.west) to swap the operator in order to truly clamp to intersection of the provided bounds and `tms.bbox`.

Sorry for all the words, happy to illustrate this better if the added test does not highlight this well enough.